subiquity/core.py

```diff
@@ -152,6 +158,33 @@ def __init__(self, opts, block_log_dir):
 
         self.install_confirmed = False
 
+    def subiquity_event(self, event):
+        if event["MESSAGE"] == "starting install":
+            if event["_PID"] == os.getpid():
+                return
+            try:
+                fcntl.flock(
+                    self.install_lock_file, fcntl.LOCK_SH | fcntl.LOCK_NB)
+            except OSError:
+                pass
+            else:
+                fcntl.flock(self.install_lock_file, fcntl.LOCK_UN)
+                return
+            from subiquity.ui.views.installprogress import (
+                InstallRunning,
+                )
+            self.install_lock_file.seek(0)
+            tty = self.install_lock_file.read().strip()
+            self.install_running = InstallRunning(self.ui.body, self, tty)
+            self.ui.body.show_stretchy_overlay(self.install_running)
+            schedule_task(self._hide_install_running())
+
+    async def _hide_install_running(self):
+        await run_in_thread(fcntl.flock, self.install_lock_file, fcntl.LOCK_SH)
+        fcntl.flock(self.install_lock_file, fcntl.LOCK_UN)
```

Comment:
We lock and unlock immediately?

Reply:
Yes, the lock attempt blocks until the install is finished.